hack/generator/pkg/astmodel/armconversion/convert_from_arm_function_builder.go

```diff
@@ -45,8 +46,11 @@ func newConvertFromARMFunctionBuilder(
 			isSpecType:            c.isSpecType,
 			codeGenerationContext: codeGenerationContext,
 		},
+		typeConversionBuilder: astmodel.NewConversionFunctionBuilder(c.idFactory, codeGenerationContext),
 	}
 
+	result.typeConversionBuilder.AddConversionHandlers(result.convertComplexTypeNameProperty)
+
 	result.propertyConversionHandlers = []propertyConversionHandler{
```

Comment:
I initially wanted to do away with this double-nesting, (we have handlers and then one of the handlers uses the `typeConversionBuilder` which itself has handlers), but actually it makes sense...

The upper handler (`propertyConversionHandlers`) is about determining which properties are involved: given a property on the destination type it determines which property (if any) on the source type will be converted to the destination.
The "inner" handler `typeConversionBuilder` is about determining how to convert between two types: given a source type and a destination type, figure out how to make an assignment. It has no knowledge of broader object structure of "other" properties.

Reply:
Done